lib/xml/parser.js

```diff
@@ -121,39 +121,71 @@ function parseEvents(xml) {
                         // Handle this next expanded occurence
                         const nextEvent = icalEvent.getOccurrenceDetails(nextOccuranceTime);
 
-                        console.log("==> ", nextOccuranceTime.toString());
+                        data = {};
 
                         if (modifiedOccurences.length === 0) {
                             // No events have been modified
+                            const dtstart = nextEvent.startDate;
+                            const dtend = nextEvent.endDate;
+
                             data.title = nextEvent.item.summary;
                             data.uid = nextEvent.item.uid;
                             data.location = nextEvent.item.location;
                             data.description = nextEvent.item.description;
-                            data.start = nextEvent.startDate.toString();
-                            data.end = nextEvent.endDate.toString();
-                            data.createdAt = vevent.getFirstPropertyValue("created").toICALString();
+                            data.start = new Date(dtstart);
+                            data.end = new Date(dtend);
+                            data.duration = dtend.subtractDate(dtstart);
+                            data.type = { recurring: true, edited: false };
+                            data.createdAt = new Date(vevent.getFirstPropertyValue("created"));
+
+                            formatted.push({
+                                ics: event["d:href"][0],
+                                etag,
+                                data
+                            });
                         } else {
                             // There are modified occurences
                             if (isModified(nextOccuranceTime, modifiedOccurences)) {
                                 // This is the event that has been modied
                                 const key = getModifiedOccuranceKey(nextOccuranceTime, modifiedOccurences) || 0;
+                                const dtstart = vevents[key].getFirstPropertyValue("dtstart");
```

Comment:
The code becomes very long here and do not really understand what is going on. Wouldn't it be nicer if you wrap the code in your if and else branches into self speaking functions?